kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -510,6 +510,11 @@ impl<'tcx> GotocCtx<'tcx> {
             }
             Variants::Multiple { tag, tag_encoding, .. } => match tag_encoding {
                 TagEncoding::Direct => {
+                    let e = if ty.is_generator() {
```

Comment:
Explain what's going on here

Reply:
Added a comment (in the factored out function).